src/aleph/vm/pool.py

```diff
@@ -202,31 +200,43 @@ def forget_vm(self, vm_hash: ItemHash) -> None:
         except KeyError:
             pass
 
+    def _schedule_forget_on_stop(self, execution: VmExecution):
+        """Create a task that will remove the VM from the pool after it stops."""
+
+        async def forget_on_stop(stop_event: asyncio.Event):
+            await stop_event.wait()
+            self.forget_vm(execution.vm_hash)
+
+        _ = asyncio.create_task(forget_on_stop(stop_event=execution.stop_event))
+
     async def _load_persistent_executions(self):
         """Load persistent executions from the database."""
         saved_executions = await get_execution_records()
         for saved_execution in saved_executions:
-            # Prevent to load the same execution twice
-            if self.executions.get(saved_execution.vm_hash):
+            vm_hash = ItemHash(saved_execution.vm_hash)
+
+            if vm_hash in self.executions:
+                # The execution is already loaded, skip it
                 continue
 
             vm_id = saved_execution.vm_id
             message_dict = json.loads(saved_execution.message)
-            original_dict = json.loads(saved_execution.original_message)
+
```

Comment:
Why remove this one? It is supose to be needed, no?

Reply:
Unused variable warning in my IDE